Sources/Datadog/Core/DD/CoreLogger.swift

```diff
@@ -0,0 +1,98 @@
+    private let currentVerbosityLevel: () -> LogLevel?
+
+    init(
+        dateProvider: DateProvider,
+        timeZone: TimeZone,
+        printFunction: @escaping (String) -> Void ,
+        verbosityLevel: @escaping () -> LogLevel?
+    ) {
+        self.dateProvider = dateProvider
+        self.dateFormatter = presentationDateFormatter(withTimeZone: timeZone)
+        self.printFunction = printFunction
+        self.currentVerbosityLevel = verbosityLevel
+    }
+
+    func debug(_ message: @autoclosure () -> String, error: Error? = nil) {
+        guard let verbosityLevel = currentVerbosityLevel(), LogLevel.debug.rawValue >= verbosityLevel.rawValue else {
+            return // if no `Datadog.verbosityLevel` is set or it is set above this level
+        }
+
+        print(message: message(), error: error, emoji: "")
+    }
+
+    func warn(_ message: @autoclosure () -> String, error: Error? = nil) {
+        guard let verbosityLevel = currentVerbosityLevel(), LogLevel.warn.rawValue >= verbosityLevel.rawValue else {
+            return // if no `Datadog.verbosityLevel` is set or it is set above this level
+        }
+
+        print(message: message(), error: error, emoji: "⚠️")
+    }
+
+    func error(_ message: @autoclosure () -> String, error: Error? = nil) {
+        guard let verbosityLevel = currentVerbosityLevel(), LogLevel.error.rawValue >= verbosityLevel.rawValue else {
+            return // if no `Datadog.verbosityLevel` is set or it is set above this level
+        }
+
+        print(message: message(), error: error, emoji: "🔥")
+    }
+
+    func critical(_ message: @autoclosure () -> String, error: Error? = nil) {
+        guard let verbosityLevel = currentVerbosityLevel(), LogLevel.critical.rawValue >= verbosityLevel.rawValue else {
+            return // if no `Datadog.verbosityLevel` is set or it is set above this level
+        }
+
+        print(message: message(), error: error, emoji: "⛔️")
+    }
+
+    // MARK: - Private
+
+    private func print(message: @autoclosure () -> String, error: Error?, emoji: String) {
+        var log = buildMessageString(message: message(), emoji: emoji)
+
+        if let error = error {
+            log += "\n\nError details:\n\(buildErrorString(error: error))"
+        }
+
+        printFunction(log)
+    }
+
+    private func buildMessageString(message: @autoclosure () -> String, emoji: String) -> String {
+        let prefix = CoreLogger.prefix
+        let time = dateFormatter.string(from: dateProvider.currentDate())
+
+        if !emoji.isEmpty {
+            return "\(prefix)\(time) \(emoji) \(message())"
+        } else {
+            return "\(prefix)\(time) \(message())"
+        }
+    }
+
+    private func buildErrorString(error: Error) -> String {
+        let dderror = DDError(error: error)
+        return """
+        → type: \(dderror.type)
+        → message: \(dderror.message)
+        → stack: \(dderror.stack)
+        """
+    }
```

Comment:
This is a new thing - if `error` object is passed, the logger will format it and print nice description in the console. Previously errors were ignored if not embedded in the `message`.